src/System.ObjectModel/ref/System.ObjectModel.cs

```diff
@@ -44,6 +46,15 @@ public partial class ObservableCollection<T> : System.Collections.ObjectModel.Co
         protected virtual void OnPropertyChanged(System.ComponentModel.PropertyChangedEventArgs e) { }
         protected override void RemoveItem(int index) { }
         protected override void SetItem(int index, T item) { }
+        #region Range functionality
+        public void AddRange(System.Collections.Generic.IEnumerable<T> collection) { }
+        public void InsertRange(int index, System.Collections.Generic.IEnumerable<T> collection) { }
+        public void RemoveRange(int index, int count) { }
+        public void RemoveRange(System.Collections.Generic.IEnumerable<T> collection) { }
+        public void ReplaceRange(System.Collections.Generic.IEnumerable<T> collection) { }
+        public void ReplaceRange(System.Collections.Generic.IEnumerable<T> collection, System.Collections.Generic.IEqualityComparer<T> comparer) { }
```

Comment:
The last 3 APIs differ from originally approved ones. Why?

Reply:
I probably missed the approved API.
Besides, I thought why not add them and let you guys consider keeping it in, since they are very useful, especially the `ReplaceRange`.

Reply:
I probably missed the approved API.
Besides, I thought why not add them and let you guys consider keeping it in, since they are very useful, especially the `ReplaceRange`.
Please refer to the [OP](https://github.com/dotnet/corefx/pull/26482#issue-290213229) to see the docs for each of the added methods.

Reply:
Technically speaking: If there are additional changes to the API surface, we will need to bring it back to API approval - every single API has to go through there ;).
However, it doesn't prevent us to look at the remaining parts of the PR and check the overall semantics - as it may spark additional discussion about the API and its shape.

Reply:
Sounds good.

Reply:
Sounds good.
See [this](https://github.com/dotnet/corefx/issues/10752#issuecomment-359197102).